yt_dlp/extractor/bilibili.py

```diff
@@ -139,283 +129,272 @@ class BiliBiliIE(InfoExtractor):
-
-            info.update({
-                'id': video_id,
-                'duration': float_or_none(durl.get('length'), 1000),
-                'formats': formats,
-                'http_headers': {
-                    'Referer': url,
-                },
-            })
-            break
 
-        self._sort_formats(formats)
-
-        title = self._html_search_regex((
-            r'<h1[^>]+title=(["])(?P<content>[^"]+)',
-            r'<h1[^>]+title=([\'])(?P<content>[^\']+)',
-            r'(?s)<h1[^>]*>(?P<content>.+?)</h1>',
-            self._meta_regex('title')
-        ), webpage, 'title', group='content', fatal=False)
-
-        # Get part title for anthologies
-        if page_id is not None:
-            # TODO: The json is already downloaded by _extract_anthology_entries. Don't redownload for each video.
-            part_info = traverse_obj(self._download_json(
-                f'https://api.bilibili.com/x/player/pagelist?bvid={bv_id}&jsonp=jsonp',
-                video_id, note='Extracting videos in anthology'), 'data', expected_type=list)
-            title = title if len(part_info) == 1 else traverse_obj(part_info, (int(page_id) - 1, 'part')) or title
-
-        description = self._html_search_meta('description', webpage)
-        timestamp = unified_timestamp(self._html_search_regex(
-            r'<time[^>]+datetime="([^"]+)"', webpage, 'upload time',
-            default=None) or self._html_search_meta(
-            'uploadDate', webpage, 'timestamp', default=None))
-        thumbnail = self._html_search_meta(['og:image', 'thumbnailUrl'], webpage)
-
-        # TODO 'view_count' requires deobfuscating Javascript
-        info.update({
-            'id': f'{video_id}_part{page_id or 1}',
-            'cid': cid,
-            'title': title,
-            'description': description,
-            'timestamp': timestamp,
-            'thumbnail': thumbnail,
-            'duration': float_or_none(video_info.get('timelength'), scale=1000),
-        })
-
-        uploader_mobj = re.search(
-            r'<a[^>]+href="(?:https?:)?//space\.bilibili\.com/(?P<id>\d+)"[^>]*>\s*(?P<name>[^<]+?)\s*<',
-            webpage)
-        if uploader_mobj:
-            info.update({
-                'uploader': uploader_mobj.group('name').strip(),
-                'uploader_id': uploader_mobj.group('id'),
-            })
-
-        if not info.get('uploader'):
-            info['uploader'] = self._html_search_meta(
-                'author', webpage, 'uploader', default=None)
+            self._sort_formats(formats)
 
-        top_level_info = {
-            'tags': traverse_obj(self._download_json(
-                f'https://api.bilibili.com/x/tag/archive/tags?aid={video_id}',
-                video_id, fatal=False, note='Downloading tags'), ('data', ..., 'tag_name')),
-        }
-
-        info['subtitles'] = {
-            'danmaku': [{
+            info_fmt = {
+                'formats': formats,
+            }
+        else:
```

Comment:
There is unnecessary amounts of nested logic

```py
        info = {'formats': []}
        for video in traverse_obj(play_info, ('dash', 'video')) or []:
            ...
        for audio in traverse_obj(play_info, ('dash', 'audio')) or []:
            ...
        if not info['formats']:
            ...
```

Reply:
en, Current it use videos_is_missing as feature to detect if it's old format,

if change to these, empty videos will also get into parse_old_flv_formats, which is not good for debuging future schema changes.

Reply:
I change the check little to make it more clear.

Reply:
My suggestion still applies though
```py
        info = {'formats': []}
        for video in traverse_obj(play_info, ('dash', 'video')) or []:
            ...
        for audio in traverse_obj(play_info, ('dash', 'audio')) or []:
            ...
        if not info['formats']:
            if 'dash' in play_info:
                self.report_warning(...)
            ...
```
The point is that this will leave most of the code less indented than it currently is and so more readable

Reply:
ok

Reply:
note that audios is used in video parse